doc/asciidoc/consumer/configuration.adoc

```diff
@@ -23,11 +25,47 @@ streams.check.apoc.timeout=<ms to await for APOC being loaded, default -1 skip t
 streams.check.apoc.interval=<ms interval awaiting for APOC being loaded, default 1000>
 streams.cluster.only=<boolean value that prevent for starting in single instances usefull when you're performing backup/restore ops, default false>
 streams.check.writeable.instance.interval=<In a cluster environment is the inveral used by the plugin to check if there was a leader re-election, in order to resume the stream process in from the last committed record in the topic. In a single instance is the interval used to check if the database became available, default 180000>
+streams.sink.poll.interval=<The delay interval between poll cycles, default 0>
 ----
 
 [NOTE]
 
 if `streams.cluster.only` is set to true, streams will refuse to start in single instance mode,
 or when run in the context of the backup operation. This is an important safety guard to ensure that operations do not occur in unexpected situations for production deploys
 
-See the https://kafka.apache.org/documentation/#brokerconfigs[Apache Kafka documentation] for details on these settings.
\ No newline at end of file
+See the https://kafka.apache.org/documentation/#brokerconfigs[Apache Kafka documentation] for details on these settings.
+
+==== Custom Kafka Configurations
+
+In this section we describe the meaning of specific Neo4j streams Kafka configurations
+
+===== `kafka.streams.async.commit`
+
+If `kafka.enable.auto.commit=false` this property allows you to manage how to commit the messages to the topic.
+
+Possible values:
+
+* `false` (default) under-the-hood we use the Kafka Consumer `commitSync` method
+* `true` under-the-hood we use the Kafka Consumer `commitAsync` method
+
+====== `commitSync` VS `commitAsync`
+
+`commitSync` is a synchronous commits and will block until either the commit
+succeeds or an unrecoverable error is encountered (in which case it is thrown
+to the caller).
+
+That means, the `commitSync` is a **blocking** method with an interal retry mechanism,
+that can affect the performance of the ingestion because a new batch of messages
+will be processed only when the commit ended.
+
+On the other hand `commitAsync` is an asynchronous call (so it will not block)
+and does not provide an internal retry mechanism.
+
+.Trade-offs: latency vs. data consistency
```

Comment:
I want to make an argument that the default should be async commit, not sync.

1. Async is faster, because we don't have to wait for word back from Kafka
2. When an error occurs in sync commit, we can't really do much about it other than log it.
3. Most user complaints have come from throughput speed, not from commit semantics.  In fact, it's pretty usual for kafka clients to be "fire and forget".   There are of course exceptions.

The only situation I can think about where commitSync is better (after reading the docs) is when you're making a complex set of dependent changes.  For example in a CDC scenario, it may be very important that kafka got the first set of CDC changes before publishing the next set, if "order of application" matters.  If you commit async, I guess there is a risk that commit 2 could succeed before commit 1 and that could be bad.

But that's all in the direction of pushing data Neo4j -> Kafka.

In the case of consuming from kafka, is the following scenario possible:

* Batch 1 comes into Neo4j (finishing with offset 10 let's say).   CommitAsync for "A" starts.
* Batch 2 comes into Neo4j (finishing with offset 20 let's say).  CommitAsync for "B" starts.
* Commit for B finishes
* Commit for A finishes
* (Kafka now thinks our partition offset is 10, when it's really 20)
* Next poll pulls old data we've already seen

Is that possible?

Reply:
From the official documentation seems that is not a problem:
https://docs.confluent.io/2.0.0/clients/consumer.html#asynchronous-commits
That use case in mi knowledge can happen only when a new consumer with the same `group.id` (ie. from a different threat) starts to consume the data, but it's a bad practice.

Reply:
I used the default `false` only for backwards compatibility. WDTY?

Reply:
I think I need to write more docs into the architecture section, specific to performance tuning.  It seems like keeping it sync is the best way to not suprise anyone, but now that you have the data to indicate that the performance is better with commitAsync, I think a lot of people will be interested in that.

Reply:
Yes totally